pyiron_ontology/parser.py

```diff
@@ -165,3 +174,41 @@ def validate_values(graph):
                             (instance, on_property, some_values_from)
                         )
     return missing_triples
+
+
+def parse_workflow(
+    workflow,
+    NS,
+    graph=None,
+    inherit_properties=True,
+    hasNode=None,
+    hasSourceFunction=None,
+    hasUnits=None,
+    inheritsPropertiesFrom=None,
+):
+    if hasNode is None:
+        hasNode = NS.hasNode
+    if hasSourceFunction is None:
+        hasSourceFunction = NS.hasSourceFunction
+    if hasUnits is None:
+        hasUnits = NS.hasUnits
+    if inheritsPropertiesFrom is None:
+        inheritsPropertiesFrom = NS.inheritsPropertiesFrom
+    if graph is None:
+        graph = Graph()
+    workflow_label = NS[workflow.label]
+    graph.add((workflow_label, RDFS.label, Literal(workflow.label)))
+    for value in workflow.children.values():
+        data = get_inputs_and_outputs(value)
```

Comment:
In the medium term we want this to work for all nodes (or at least all nodes that could be children of a macro or workflow -- i.e. nodes with their IO defined at the class-level) -- but right now it works exclusively for `Function` nodes. We can delay the full functionality, but let's already have the failure happen cleanly with something like an `isisntance` and a very clear error message. That could be here, but inside `get_inputs_and_outputs` probably makes even more sense.